datasets/ru/egrul/crawler.py

```diff
@@ -446,6 +447,52 @@ def parse_address(context: Context, entity: Entity, el: Element) -> None:
     entity.add("address", address)
 
 
+def compile_abbreviations(context) -> List[Tuple[str, re.Pattern, List[str]]]:
+    """
+    Load abbreviations and compile regex patterns from the YAML config.
+    Returns:
+    List[Tuple[str, re.Pattern, List[str]]]: A list of tuples containing canonical abbreviations
+    and their compiled regex patterns for substitution.
+    """
+    yaml = context.dataset.config.get("types")
+    abbreviations = []
+    for canonical, phrases in yaml.items():
+        # we want to match the whole word and allow for ["] or ['] at the beginning
+        for phrase in phrases:
+            phrase_pattern = rf"^[ \"']?{re.escape(phrase)}"
+
+            compiled_pattern = re.compile(phrase_pattern, re.IGNORECASE)
+            # Append the canonical form, compiled regex pattern, and sorted phrases to the list
+            abbreviations.append((canonical, compiled_pattern, phrase))
+
+    abbreviations.sort(key=lambda x: len(x[2]), reverse=True)
+    return abbreviations
+
+
+def substitute_abbreviations(
+    name: str, abbreviations: List[Tuple[str, re.Pattern, List[str]]]
+) -> Tuple[str, Optional[str]]:
+    """
+    Substitute abbreviations in the name using the compiled regex patterns.
+    :param name: The input name where abbreviations should be substituted.
+    :param abbreviations: A list of tuples with canonical abbreviations, regex patterns,
+                          and original phrases.
+    :return: The modified text with substitutions applied.
+    """
+
+    # Iterate over all abbreviation groups
+    for canonical, pattern, phrases in abbreviations:
```

Comment:
I'd call this `regex` rather than `pattern` since a "pattern" is what gets compiled

```suggestion
    for canonical, regex, phrases in abbreviations:
```

Reply:
total nitpick, but FYI